token/program-2022/src/extension/confidential_mint_burn/processor.rs

```diff
@@ -0,0 +1,405 @@
+#[cfg(feature = "zk-ops")]
+use spl_token_confidential_transfer_ciphertext_arithmetic as ciphertext_arithmetic;
+use {
+    crate::{
+        check_program_account,
+        error::TokenError,
+        extension::{
+            confidential_mint_burn::{
+                instruction::{
+                    BurnInstructionData, ConfidentialMintBurnInstruction, InitializeMintData,
+                    MintInstructionData, RotateSupplyElGamalPubkeyData,
+                    UpdateDecryptableSupplyData,
+                },
+                verify_proof::{verify_burn_proof, verify_mint_proof},
+                ConfidentialMintBurn,
+            },
+            confidential_transfer::{ConfidentialTransferAccount, ConfidentialTransferMint},
+            BaseStateWithExtensions, BaseStateWithExtensionsMut, PodStateWithExtensionsMut,
+        },
+        instruction::{decode_instruction_data, decode_instruction_type},
+        pod::{PodAccount, PodMint},
+        processor::Processor,
+    },
+    solana_program::{
+        account_info::{next_account_info, AccountInfo},
+        entrypoint::ProgramResult,
+        msg,
+        program_error::ProgramError,
+        pubkey::Pubkey,
+    },
+    solana_zk_sdk::{
+        encryption::pod::{auth_encryption::PodAeCiphertext, elgamal::PodElGamalPubkey},
+        zk_elgamal_proof_program::proof_data::{
+            CiphertextCiphertextEqualityProofContext, CiphertextCiphertextEqualityProofData,
+        },
+    },
+    spl_token_confidential_transfer_proof_extraction::instruction::verify_and_extract_context,
+};
+
+/// Processes an [InitializeMint] instruction.
+fn process_initialize_mint(accounts: &[AccountInfo], data: &InitializeMintData) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+    let mint_info = next_account_info(account_info_iter)?;
+
+    check_program_account(mint_info.owner)?;
+
+    let mint_data = &mut mint_info.data.borrow_mut();
+    let mut mint = PodStateWithExtensionsMut::<PodMint>::unpack_uninitialized(mint_data)?;
+    let mint_burn_extension = mint.init_extension::<ConfidentialMintBurn>(true)?;
+
+    mint_burn_extension.authority = data.authority;
```

Comment:
To stop people from shooting themselves in the foot, can this return an error if `data.authority == None`?